### PR TITLE
Return py42 response on all writes

### PR DESCRIPTION
**Pull request by unparalleled-js**
### Description of Change ###

- functions now return `Py42Response` objects as expected:
    - `sdk.detectionlists.update_user_notes()`
    - `sdk.detectionlists.add_user_risk_tags()`
    - `sdk.detectionlists.remove_user_risk_tags()`
    - `sdk.detectionlists.add_user_cloud_alias()`
    - `sdk.detectionlists.remove_user_cloud_alias()`
- `sdk.archive.get_all_org_cold_storage_archives()` now actually uses parameters `include_child_orgs`, `sort_key` and `sort_dir`

Correct other, minor documentation strings too.

### Issues Resolved ###

- closes #INTEG-1091

### Testing Procedure ###
* Call any of the methods (listed in CHANGELOG) and make sure you get a Py42Response object as a response. Before, it was just None.

### PR Checklist ###
Did you remember to do the below?

- [x] Add unit tests to verify this change
- [x] Add an entry to CHANGELOG.md describing this change
- [n/a] Add docstrings for any new public parameters / methods / classes

### CODE REVIEW

File: src/py42/modules/archive.py [hunk 1]
```diff
@@ -146,4 +146,9 @@ def get_all_org_cold_storage_archives(
             generator: An object that iterates over :class:`py42.response.Py42Response` objects
             that each contain a page of cold storage archive information.
         """
-        return self._archive_client.get_all_org_cold_storage_archives(org_id)
+        return self._archive_client.get_all_org_cold_storage_archives(
```
Comment (unparalleled-js) [inline src/py42/modules/archive.py:149]: The params did nothing before

File: tests/modules/test_detectionlists.py [hunk 2]
```diff
@@ -75,7 +75,17 @@ def test_update_user_notes_calls_user_client_with_expected_values(
...
-    def test_add_user_risk_tag_calls_user_client_with_expected_values(
+    def test_update_user_notes_returns_response(
+        self, mock_microservice_client_factory, mock_detection_list_user_client
+    ):
+        mock_microservice_client_factory.get_detection_list_user_client.return_value = (
+            mock_detection_list_user_client
+        )
+        module = DetectionListsModule(mock_microservice_client_factory)
+        response = module.update_user_notes(TEST_USER_ID, "newnotes")
+        assert response
```
Comment (unparalleled-js) [inline tests/modules/test_detectionlists.py:86]: type is MagicMock in this case, but before was None, so these tests catch the error

**Comment by unparalleled-js** [comment]
Also, I looked for other methods like this but couldn't find any

File: tests/modules/test_detectionlists.py [hunk 3]
```diff
@@ -123,6 +163,16 @@ def test_remove_user_cloud_alias_calls_user_client_with_expected_values(
...
 
+    def test_remove_user_cloud_alias_returns_response(
+        self, mock_microservice_client_factory, mock_detection_list_user_client
+    ):
+        mock_microservice_client_factory.get_detection_list_user_client.return_value = (
+            mock_detection_list_user_client
+        )
+        module = DetectionListsModule(mock_microservice_client_factory)
+        response = module.remove_user_cloud_alias(TEST_USER_ID, "oldalias")
+        assert response
```
Comment (alanag13) [inline tests/modules/test_detectionlists.py:174]: adopting this approach for testing each call we make in general will probably be a good idea going forward.

Comment (unparalleled-js) [reply tests/modules/test_detectionlists.py:174]: If you want, I add them all now! But that would be a lot

File: src/py42/modules/archive.py (hunk 1 above)
Comment (alanag13) [reply src/py42/modules/archive.py:149]: good find, this was definitely my fault :-/

File: tests/modules/test_detectionlists.py (hunk 3 above)
Comment (alanag13) [reply tests/modules/test_detectionlists.py:174]: no no, wasn't suggesting that, just putting thoughts out there

**Comment by alanag13** [comment]
I want to retest `get_all_org_cold_storage_archives()` for safety's sake, but this otherwise looks good, will approve once that's confirmed as good.

**Pull request merged**